R/util-checkSpec.R

```diff
@@ -0,0 +1,66 @@
+# Check that all data.frames and columns in the spec are present in the data
+#' Function to check if the data and spec are compatible
+#'
+#' This function checks if the data and spec are compatible by comparing the data.frames
+#' and columns in the spec with the data.
+#'
+#' @param lData A list of data.frames.
+#' @param lSpec A list s specifying the expected structure of the data.
+#'
+#' @return This function does not return any value. It either prints a message indicating
+#' that all data.frames and columns in the spec are present in the data, or throws an error
+#' if any data.frame or column is missing.
+#'
+#' @examples
+#' lData <- list(reporting_groups = gsm::reportingGroups, reporting_results = gsm::reportingResults)
+#' lSpec <- list(
+#'   reporting_groups = list(  
+#'       GroupID = "required", 
+#'       GroupLevel = "required",
+#'       Param = "required",
+#'       Value = "required",
+#'   ),
+#'   reporting_results = list(
+#'       GroupID = "required",
+#'       GroupLevel = "required",
+#'       Numerator = "required",
+#'       Denominator = "required"
+#'   )
+#' CheckSpec(lData, lSpec) # Prints message that everything is found
+#'
+#' lSpec$reporting_groups$NotACol <- "required"
+#' check_spec(lData, lSpec) # Throws error that NotACol is missing
+#'
+#' @export
+#' 
+CheckSpec <- function(lData, lSpec) {
+    # Check that all data.frames in the spec are present in the data
+    lSpecDataFrames <- names(lSpec)
+    lDataFrames <- names(lData)
+    if (!all(lSpecDataFrames %in% lDataFrames)) {
+        MissingSpecDataFrames <- lSpecDataFrames[!lSpecDataFrames %in% lDataFrames]
+        cli_alert_danger("Not all data.frames in the spec are present in the data, missing data.frames are: {MissingSpecDataFrames}")
```

Comment:
```suggestion
        cli::cli_abort(c(
            "{.arg lData} must contain all data.frames in {.arg lSpec}.",
            i = "Missing data.frames: {MissingSpecDataFrames}"
        ))
```
`cli_alert_danger()` doesn't actually error, it just tells the user that something bad happened. If we're checking a spec, we should stop with an actual error.

I prefer [tidyverse style for error messages](https://style.tidyverse.org/error-messages.html), which is basically a short, clear problem statement on one line, and then details. I included that format here, feel free to ignore that part of the suggestion if you want to stick with the one-liner.